components/calendar/src/chinese_based.rs

```diff
@@ -232,76 +414,165 @@ impl<C: ChineseBased + CalendarArithmetic> ChineseBasedDateInner<C> {
         }
     }
 
-    /// Get a ChineseBasedDateInner from a fixed date
-    ///
-    /// Months are calculated by iterating through the dates of new moons until finding the last month which
-    /// does not exceed the given fixed date. The day of month is calculated by subtracting the fixed date
-    /// from the fixed date of the beginning of the month.
+    /// Given a 1-indexed chinese extended year, fetch its data from the cache.
     ///
-    /// The calculation for `elapsed_years` and `month` in this function are based on code from _Calendrical Calculations_ by Reingold & Dershowitz.
-    /// Lisp reference code: https://github.com/EdReingold/calendar-code2/blob/main/calendar.l#L5414-L5459
-    pub(crate) fn chinese_based_date_from_fixed(date: RataDie) -> ChineseBasedDateInner<C> {
-        let (first_day_of_year, next_solstice) = Self::new_year_on_or_before_fixed_date(date, None);
-        let next_new_year =
-            Self::new_year_on_or_before_fixed_date(first_day_of_year + 400, Some(next_solstice)).0;
-        let year_float = libm::floor(
-            1.5 - 1.0 / 12.0 + ((first_day_of_year - C::EPOCH) as f64) / MEAN_TROPICAL_YEAR,
-        );
-        let year_int = i64_to_i32(year_float as i64);
-        debug_assert!(
-            matches!(year_int, I32Result::WithinRange(_)),
-            "Year should be in range of i32"
-        );
-        let year = year_int.saturate();
-        let new_moon = Self::new_moon_before((date + 1).as_moment());
-        let month_i64 =
-            libm::round((new_moon - first_day_of_year) as f64 / MEAN_SYNODIC_MONTH) as i64 + 1;
+    /// If the actual year data that was fetched is for a different year, update the getter year
+    fn get_compiled_data_for_year_helper(
+        date: RataDie,
+        getter_year: &mut i32,
```

Comment:
Nit (optional): I still think this function should take only a `RataDie`, and the extended year can be retrieved by the caller from the return value

Reply:
^ this hasn't been responded to

Reply:
> ^ this hasn't been responded to

Yes, it was an optional nit. The return value does not contain the extended year.